src/woo_publications/publications/admin.py

```diff
@@ -303,6 +303,17 @@ def delete_queryset(
                     )
                 )
 
+    def save_related(
+        self,
+        request: HttpRequest,
+        form: forms.Form,
+        formsets: forms.BaseModelFormSet,
+        change: bool,
+    ):
+        super().save_related(request, form, formsets, change)
+        if not change:
+            form.instance.apply_retention_policy()  # pyright: ignore[reportAttributeAccessIssue]
```

Comment:
Better is:

```suggestion
        if not change:
            assert isinstance(form.instance, Publication)
            form.instance.apply_retention_policy()
```

Reply:
I did try that but doesn't fix the error `Cannot access attribute "instance" for class "Form"
    Attribute "instance" is unknown (reportAttributeAccessIssue)`